botorch/generation/gen.py

```diff
@@ -212,18 +212,22 @@ def f(x):
         options={k: v for k, v in options.items() if k not in ["method", "callback"]},
     )
 
-    if "success" not in res.keys():
-        warnings.warn(
-            "Optimization failed within `scipy.optimize.minimize` with no "
-            "status returned to `res.`",
-            OptimizationWarning,
-        )
+    if "success" not in res.keys() or "status" not in res.keys():
+        with warnings.catch_warnings():
+            warnings.simplefilter("always", category=OptimizationWarning)
```

Comment:
I'm not a fan of having these two lines repeated so many times in the code. Probably not a concern here, but this also prevents the user from suppressing these warnings if they wanted to.

Reply:
I don't think it does -- I suppressed these in a few tests where they weren't relevant.

Reply:
Hmm, this still raises the warning for me:
```
import warnings

warnings.simplefilter("ignore")
with warnings.catch_warnings():
    warnings.simplefilter("always")
    warnings.warn("dummy warning", RuntimeWarning)
```

Reply:
Oh, yeah, good catch. I was using `warnings.catch_warnings(record=True) as ws` to prevent the warning from being written to stdout, but it still seems to be added to `ws` regardless of any `ignore` filtering.